src/middlewares/user.middleware.js

```diff
@@ -1,12 +1,24 @@
 import { User } from '../database/models/index';
 
 export const checkUserExists = async (req, res, next) => {
-  const { email } = req.body;
-  const userInDb = await User.findOne({
-    where: { email: email },
-  });
-  if (userInDb) {
-    return res.status(409).json({ message: 'Email already exists' });
-  }
-  next();
+	const { email } = req.body;
+	const userInDb = await User.findOne({
+		where: { email: email },
+	});
+	if (userInDb) {
+		return res.status(409).json({ message: 'Email already exists' });
+	}
+	next();
+};
+
+export const checkUserEmailVerified = async (req, res, next) => {
+	const { email } = req.body;
+	const user = await User.findOne({ where: { email: email } });
+	if (!user) {
+		return res.status(400).json({ error: 'Wrong credentials' });
+	}
+	if (!user.verified) {
+		return res.status(400).json({ error: 'User email is not verified' });
+	}
+	next();
 };
```

Comment:
Let's have two separate middleware:
- Check if the user exist in the DB
- Check if the user is verified